pkgs/by-name/ar/arduino-ide/package.nix

```diff
@@ -0,0 +1,38 @@
+{ appimageTools
+, fetchurl
+, lib
+}:
+
+let
+  pname = "arduino-ide";
+  version = "2.2.1";
+
+  src = fetchurl {
+    url = "https://github.com/arduino/arduino-ide/releases/download/${version}/arduino-ide_${version}_Linux_64bit.AppImage";
+    hash = "sha256-77uS/3ean3dWG/vDHG+ry238hiJlYub7H03f15eJu+I=";
+  };
+
+  appimageContents = appimageTools.extractType2 { inherit pname version src; };
+in
+appimageTools.wrapType2 {
+  inherit pname version src;
+
+  extraInstallCommands = ''
+    mv $out/bin/{${pname}-${version},${pname}}
+
+    install -Dm444 ${appimageContents}/${pname}.desktop -t $out/share/applications/
+    install -Dm444 ${appimageContents}/${pname}.png -t $out/share/pixmaps/
+  '';
+
+  extraPkgs = pkgs: with pkgs; [ libsecret ];
+
+  meta = with lib; {
+    description = "Open-source electronics prototyping platform";
+    homepage = "https://www.arduino.cc/en/software";
+    changelog = "https://github.com/arduino/arduino-ide/releases/tag/${version}";
+    license = licenses.agpl3;
+    mainProgram = "arduino-ide";
+    maintainers = with maintainers; [ clerie ];
+    platforms = [ "x86_64-linux" ];
```

Comment:
note(non-blocking): I'm aware you mentioned you want to use the AppImage as a source, because building from source may be painful.
There is, however, a darwin build, too. So it might one day be worth putting in the effort to build it for both platforms.

I don't feel strongly about this, just wanted to leave the note here.
Maybe it's a good idea to add a todo comment for darwin?

Reply:
I trust people to find out that there are Darwin binaries for it. So I don't think we need an explicit comment about this.

Whoever wants to package the Darwin binaries should just open a PR then.